lib/Support/BigIntSupport.cpp

```diff
@@ -1868,7 +1864,14 @@ static OperationStatus compute(
 } // namespace
 
 uint32_t divideResultSize(ImmutableBigIntRef lhs, ImmutableBigIntRef rhs) {
-  return div_rem::getResultSize(lhs, rhs);
+  if (compare(rhs, -1) == 0 && isNegative(lhs)) {
```

Comment:
nit: `isNegative` is cheaper than `compare`.

```suggestion
  if (isNegative(lhs) && compare(rhs, -1) == 0) {
```

Reply:
Say 50% chance rhs is negative, we skip expensive check half the time. Versus expensive check everytime to skip simple check. compare has 2 ifNegative checks internally before the actual compare calc. But, unlikely to make much difference in grand scheme of division, that's why it's just a nit.

Reply:
> Say 50% chance lhs is negative, we skip expensive check half the time.

I did not benchmark those, but if `compare` is properly optimized, it is (1) not taking a constant time as you assume and (2) provides more bits of certainty per operation than `isNegative` and hence should be more performant as a check

It's most likely not relevant here though for the main performance benefit

Reply:
https://github.com/facebook/hermes/pull/1856